### PR TITLE
Add Intel apx, avx10, keylocker

### PR DESCRIPTION
**Pull request by klauspost**
Only KEYLOCKER has been released in the wild, AFAIK.

`AVX10_512` is tricky, since it depends on which core is being tested.

It seems like code will need to do additional checks, since it may have been reassigned by the runtime.

### CODE REVIEW

**Comment by klauspost** [comment]
Ping @mythi - if they have the time to review.

**Review by mythi — COMMENTED** [review]
I'll see if I can get this tested. One minor comment until that time.

File: mockcpu_test.go [hunk 1]
```diff
@@ -197,6 +197,9 @@ func TestMocks(t *testing.T) {
 			t.Log("L3 Cache:", CPU.Cache.L3, "bytes")
 			t.Log("Hz:", CPU.Hz, "Hz")
 			t.Log("Boost:", CPU.BoostFreq, "Hz")
+			if CPU.AVX10Level > 0 {
```
[the remark this answers is not in the document]
Comment (klauspost) [reply mockcpu_test.go:200]: Ah, yeah.. Forgot about that one! Thanks!

**Comment by klauspost** [comment]
I will merge for now - seems we have some time before real hardware hits :)

**Pull request merged**